src/app/services/technical-record/technical-record.service.ts

```diff
@@ -172,6 +170,23 @@ export class TechnicalRecordService {
     return this.http.put<VehicleTechRecordModel>(url, body, { responseType: 'json' });
   }
 
+  calculateTechRecordToUpdate(newVehicleTechRecord: VehicleTechRecordModel, recordToArchiveStatus?: StatusCodes): TechRecordModel {
+    let newTechRecord;
+
+    if (recordToArchiveStatus === StatusCodes.PROVISIONAL) {
+      newTechRecord = newVehicleTechRecord.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.PROVISIONAL);
+    } else {
+      newTechRecord = newVehicleTechRecord.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.CURRENT);
+      if (!newTechRecord) {
+        newTechRecord = newVehicleTechRecord.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.PROVISIONAL);
+      }
+    }
+
+    if (!newTechRecord) throw new Error(`Cannot find a provisional or current to update`);
+
+    return newTechRecord;
+  }
```

Comment:
1. To not hijack the order of methods in this service, can you please move this helper method under `filterTechRecordByStatusCode()`.
2. You can simplify the logic as such:
```suggestion
  calculateTechRecordToUpdate(vehicle: VehicleTechRecordModel, status?: StatusCodes): TechRecordModel {
    let techRecord = vehicle.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.CURRENT);

    if (!techRecord || status === StatusCodes.PROVISIONAL) {
      techRecord = vehicle.techRecord.find(techRecord => techRecord.statusCode === StatusCodes.PROVISIONAL);
    }

    if (!techRecord) throw new Error(`Cannot find a provisional or current to update`);

    return techRecord;
  }
```

Reply:
This is nice! Thanks man